garak/generators/huggingface.py

```diff
@@ -59,15 +58,20 @@ class Pipeline(Generator, HFCompatible):
     generator_family_name = "Hugging Face 🤗 pipeline"
     supports_multiple_generations = True
 
-    def _set_hf_context_len(self, config):
-        if hasattr(config, "n_ctx"):
-            if isinstance(config.n_ctx, int):
-                self.context_len = config.n_ctx
-
-    def __init__(self, name, do_sample=True, generations=10, device=0):
-        self.fullname, self.name = name, name.split("/")[-1]
+    def __init__(
+        self, name="", do_sample=True, generations=10, device=0, config_root=_config
+    ):
+        self.name = name
+        self.generations = generations
+        self.do_sample = do_sample
+        self.device = device
+        self._load_config(config_root)
```

Comment:
`HFCompatible` is providing `_set_hf_context_len()` and is about extracting the the context length from the config of the transformer object after it has been instantiated.  I have removed the redundant  `_set_hf_context_len()` implementation from `Pipeline` because not all classes here extend `Pipeline`.  On the other hand `_set_hf_context_len()` could be refactored as a static utility function that extracts the length from from the instance if found and returns it to allow removal of the class.

```
@staticMethod
def _set_hf_context_len(config):
    if hasattr(config, "n_ctx"):
        if isinstance(config.n_ctx, int):
            return config.n_ctx
    return None
```

```
self._set_hf_context_len(self.generator.model.config)
```
becomes:
```
self.context_len = _set_hf_context_len(self.generator.model.config)
```

Reply:
Oh, right. Can I ask - why static?

Reply:
Simply, because it does not need to mutate the instance I think python may take some additional context from the `@staticMethod` annotation so will look closer at if it really has value here.

Reply:
Side note -- `@staticmethod` is all lower-case.